spec/controllers/two_factor_authentication/sms_opt_in_controller_spec.rb

```diff
@@ -52,36 +54,22 @@
     context 'when loaded while adding a new phone' do
       let(:user) { create(:user) }
       let(:phone) { Faker::PhoneNumber.cell_phone }
-      let(:user_session) { { unconfirmed_phone: phone } }
+      let(:opt_out_uuid) { PhoneNumberOptOut.create_or_find_with_phone(phone).uuid }
+
       before do
         stub_sign_in_before_2fa(user)
-        allow(controller).to receive(:user_session).
-          and_return(user_session)
       end
 
       it 'assigns an in-memory phone configuration' do
         expect { action }.to_not change { user.reload.phone_configurations.count }
 
-        expect(assigns[:phone_configuration].phone).to eq(phone)
-      end
-
-      context 'when user_session has both an unconfirmed phone and a phone_id' do
-        let(:user_session) do
-          {
-            unconfirmed_phone: phone,
-            phone_id: create(:phone_configuration).id,
-          }
-        end
-
-        it 'prefers the unconfirmed_phone' do
-          action
-
-          expect(assigns[:phone_configuration].phone).to eq(phone)
-        end
+        expect(assigns[:phone_configuration].formatted_phone).to eq(PhoneFormatter.format(phone))
```

Comment:
FYI I've been seeing some flakiness with this assertion.

Example: https://app.circleci.com/pipelines/github/18F/identity-idp/70137/workflows/b9cb7b04-9270-4a4f-96ee-860680ea5273/jobs/110921

```
  1) TwoFactorAuthentication::SmsOptInController#new when loaded while adding a new phone assigns an in-memory phone configuration
     Failure/Error: expect(assigns[:phone_configuration].formatted_phone).to eq(PhoneFormatter.format(phone))
     
       expected: "+14655014146"
            got: "+46 550 141 46"
     
       (compared using ==)
```

Reply:
Thanks for flagging, I will take a look

Reply:
https://github.com/18F/identity-idp/pull/5966